knotx-adapter/knotx-adapter-common/src/main/java/io/knotx/adapter/common/http/HttpClientFacade.java

```diff
@@ -154,11 +149,9 @@ private MultiMap getFilteredHeaders(MultiMap headers, List<Pattern> allowedHeade
         .collect(MultiMapCollector.toMultiMap(o -> o, headers::getAll));
   }
 
-  private Observable<ClientResponse> wrapResponse(HttpClientResponse response) {
-    return Observable.just(Buffer.buffer())
-        .mergeWith(response.toObservable())
-        .reduce(Buffer::appendBuffer)
-        .doOnNext(this::traceServiceCall)
+  private Single<ClientResponse> wrapResponse(HttpResponse<Buffer> response) {
+    return Single.just(response.body())
```

Comment:
Yes - class `HttpResponse` comes from new `vertx-web-client` and is should all the "buffering" job for us. We get full `body` here.